[Feat/Application] Admin_application_view

Kang9805 opened the pull request:
## ✅ PR 요약
- admin_application_views, test code 추가, urls 맵핑

## 📄 상세 내용
- [ ] AdminApplicationListView 신규 구현
- [ ] AdminApplicationDetailView 신규 구현
- [ ] Admin 전용 Serializer 구조 정리 및 상세 필드 추가
- [ ] application_urls.py 내부 구조 개선 및 admin_urls.py 연동
- [ ] 관리자 API 테스트 코드(test_admin_application_views.py) 추가
- [ ] 페이지네이션 임시 TODO 처리

## 📸 스크린샷 (선택)

## 📝 기타 참고 사항
- [ ] python manage.py test apps.application.tests.test_admin_application_views 로 테스트 

## 🧪 PR Checklist
- [ ] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [ ] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

Kang9805 committed:
✨ feat : 관리자용 지원서 목록/상세 조회 기능 구현

- AdminApplicationListView 신규 구현
- AdminApplicationDetailView 신규 구현
- Admin 전용 Serializer 구조 정리 및 상세 필드 추가
- application_urls.py 내부 구조 개선 및 admin_urls.py 연동
- 관리자 API 테스트 코드(test_admin_application_views.py) 추가
- 페이지네이션 임시 TODO 처리

SWWS97 commented on apps/application/serializers/admin_application_serializers.py:
```diff
@@ -55,6 +65,43 @@ class Meta:
+        read_only_fields = ["id", "name"]
+
+
+class AdminRecruitmentDetailSerializer(serializers.ModelSerializer[Application]):
```
여기 Application이 아니라 Recruitment가 되어야 할거 같아요

Kang9805 replied:
넵 확인하고 수정하겠습니다.

SWWS97 commented on apps/application/views/admin_application_views.py:
```diff
@@ -0,0 +1,123 @@
+    def get(self, request: Request) -> Response:
+
+        user = request.user
+        if not user.is_staff:
```
위에서 이미 permission_classes = [IsAdminUser] 권한 체크가 되어있어서 중복 검증으로 보여요

SWWS97 commented on apps/application/views/admin_application_views.py:
```diff
@@ -0,0 +1,123 @@
+        if applicant_nickname:
+            qs = qs.filter(applicant__nickname__icontains=applicant_nickname)
+
+        # 정렬
```
여기 위에서 SORT_MAP 만들어 놓으시고 사용안하신거 같아요

Kang9805 committed:
💡 chore : admin_application_serializer 수정

AdminRecruitmentDetailSerializer의 타입 힌트 수정(Application → Recruitment)

SWWS97 commented on apps/application/views/admin_application_views.py:
```diff
@@ -0,0 +1,123 @@
+    )
+    def get(self, request: Request, application_uuid: str) -> Response:
+        application = (
+            Application.objects.select_related("recruitment", "applicant").filter(uuid=application_uuid).first()
```
여기서 recritment.tags, recruitment.lectures 조회 할때 쿼리가 또 발생하니까 prefetch 걸어두시는게 좋을거 같습니다

Kang9805 committed:
♻️ refactor : admin application serializer의 lecture/tag 처리 방식 수정 및 mypy 오류 해결

- RecruitmentDetailSerializer에서 lectures, tags 조회 시 study_group → studylecture 연결 방식 적용
- recruitment_tags 기반 태그 조회 로직 정리
- SerializerMethodField 반환 타입을 Any로 변경하여 mypy 타입 충돌 해결
- admin application 목록/상세 조회 관련 prefetch 설정 보완

Kang9805 committed:
- 페이지네이션 전용 클래스 추가(AdminApplicationPagination)
- 테스트 전부 통과하도록 검색 및 정렬 로직 보완
- 검색 기능 통합(search: title/nickname/email)
- status 필터링을 choices 기반으로 개선

SWWS97 commented on apps/application/views/admin_application_views.py:
```diff
@@ -0,0 +1,120 @@
+        paginator = AdminApplicationPagination()
+        paginated_qs = paginator.paginate_queryset(qs, request)
+
+        serializer = AdminApplicationListSerializer(qs, many=True)
```
페이지네이션 직렬화 하시는데 위에 paginated_qs로 가져오시는데 여기서 qs는 무엇일까요?

Kang9805 replied:
수정 완료했습니다.

SWWS97 commented on apps/application/views/admin_application_views.py:
```diff
@@ -0,0 +1,120 @@
+    )
+    def get(self, request: Request, application_uuid: str) -> Response:
+
+        application = (
```
여기랑 59번째줄이 똑같은 쿼리셋인데 공통 쿼리셋 하나 만들어서 사용하시죠

Kang9805 committed:
♻️ refactor : 공통 QuerySet 분리 및 타입 힌트 추가로 코드 구조 개선

- admin application 조회 공통 쿼리셋(get_admin_application_queryset) 분리
- select_related / prefetch_related 중복 제거
- QuerySet[Application] 반환 타입 명시하여 mypy 오류 해결
- pagination 적용 시 paginated_qs로 시리얼라이저 변경
- 정렬(sort) 처리 로직 안정성 개선 (fallback 포함)
